corgidrp/detector.py

```diff
@@ -520,7 +525,161 @@ def calc_sat_fwc(emgain_arr,fwcpp_arr,fwcem_arr,sat_thresh):
+    rad = planet_rad
+    if band == '1':
+        mask=mask1
+    elif band=='4':
+        mask=mask4
+    for i in np.arange(len(residual_images)):
+        nx = np.arange(0,residual_images[i].shape[1])
+        ny = np.arange(0,residual_images[i].shape[0])
+        nxx,nyy = np.meshgrid(nx,ny)
+        nrr = np.sqrt((nxx-rad-5)**2 + (nyy-rad-5)**2)
+        nrr_copy = nrr.copy()
+        nrr_copy[nrr<rad] = residual_images[i][nrr<rad] 
+        nrr_copy[nrr>=rad] = None 
+        ymin = int(cent[i][0])
+        ymax = int(cent[i][1])
+        xmin = int(cent[i][2])
+        xmax = int(cent[i][3])
+        
+        bool_innotzero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        bool_iniszero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]==0)
+        bool_outisnotzero = np.logical_and(nrr>=rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        
+        full_residuals[ymin:ymax,xmin:xmax][bool_innotzero] = (nrr_copy[bool_innotzero] + full_residuals[ymin:ymax,xmin:xmax][bool_innotzero])/2
+        full_residuals[ymin:ymax,xmin:xmax][bool_iniszero] = nrr_copy[bool_iniszero]
+        full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero] = full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero]
+        
+        
+        
+        full_qe_resel = ndimage.convolve(full_qe, mask)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask)
+        std_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel))
+    
+    full_residuals[full_residuals==0] = None
+    nx = np.arange(0,std_resel.shape[1])
+    ny = np.arange(0,std_resel.shape[0])
+    nxx,nyy = np.meshgrid(ny,nx)
+    nrr = np.sqrt((nxx-n/2)**2 + (nyy-n/2)**2)
+    full_residuals[nrr>n_pix]= 1
+    std_resel[nrr>n_pix] = 0
+    full_qe[nrr>n_pix]= 1
+    full_qe=np.pad(full_qe, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(1))
+    full_residuals=np.pad(full_residuals, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(1))
+    std_resel=np.pad(std_resel, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(0))
+    
+    return (full_qe,full_residuals,std_resel)
+    
+    
+def create_onsky_flatfield(dataset, planet=None,band=None,up_radius=None,im_size=None,rad_mask1=None, rad_mask4=None,  planet_rad=None, n_pix=None, n_pad=None,N=None):
+    """Turn this dataset of raster scanned image frames of neptune or uranus into on-sky calibrated flat field 
+    
+        Args:
+            dataset (corgidrp.data.Dataset): a dataset of Image frames (L2a-level)
+            planet (str): neptune or uranus
+            band (str): 1 or 4
+            up_radius (int): Number of pixels on either side of centroided planet images (=55 pixels for Neptune and uranus)
+            im_size (int): x-dimension of the planet image (in pixels= 420 for the HST images)
+            N (int): Number of images to be combined for creating a matched filter (defaults to 3, can be changed)
```

Comment:
The order of the arguments in the docstring should match the order they are defined in the fucntino signature.